skbio/stats/distance/_base.py

```diff
@@ -487,6 +487,34 @@ def to_data_frame(self):
         """
         return pd.DataFrame(data=self.data, index=self.ids, columns=self.ids)
 
+    @experimental(as_of="0.5.0-dev")
+    def to_series(self):
+        """Create a ``pandas.Series`` from this ``DissimilarityMatrix``.
+
+        Returns
+        -------
+        pd.Series
+            ``pd.Series`` with IDs on index.
+
+        Examples
+        --------
+        >>> from skbio import DistanceMatrix
+        >>> dm = DistanceMatrix([[0, 1, 2],[1,0,3],[2,3,0]], ids=['a','b','c'])
+        >>> dm.to_series()
+        a  b    1.0
+           c    2.0
+        b  a    1.0
+           c    3.0
+        c  a    2.0
+           b    3.0
+        dtype: float64
+
+        """
+        df = pd.DataFrame(data=self.data, index=self.ids, columns=self.ids)
+        stacked = df.stack()
+        series = stacked[stacked > 0]
```

Comment:
The diagonal is guaranteed to be zero but there can be zeros elsewhere in the matrix so this strategy won't always work as expected. We also only want to include pairwise distances from one of the matrix triangles.

You can use `self.condensed_form()` to obtain a 1-D numpy array in condensed form, and then you can build the `pd.MultiIndex` by generating ID pairs with `itertools.combinations`. `to_series` can be implemented as follows:

```python
distances = self.condensed_form()
# `id_pairs` will not be interpreted as a `pd.MultiIndex` if it is an iterable returned by `itertools.combinations`.
id_pairs = list(itertools.combinations(self.ids, 2))
index = pd.Index(id_pairs, tupleize_cols=True)
return pd.Series(data=distances, index=index, dtype=float)
```